src/main/java/me/botsko/prism/listeners/PrismPlayerEvents.java

```diff
@@ -39,430 +48,436 @@
-				case DARK_OAK_SAPLING:
-				case CARROT:
-				case POTATO:
+    private final Prism plugin;
+    private final List<String> illegalCommands;
+    private final List<String> ignoreCommands;
+
+    /**
+     * Constructor.
+     * @param plugin Prism
+     */
+    public PrismPlayerEvents(Prism plugin) {
+        this.plugin = plugin;
+        illegalCommands = plugin.getConfig().getStringList("prism.alerts.illegal-commands.commands");
+        ignoreCommands = plugin.getConfig().getStringList("prism.do-not-track.commands");
+    }
+
+    /**
+     * Log command use.
+     *
+     * @param event PlayerCommandPreprocessEvent
+     */
+    @EventHandler(priority = EventPriority.MONITOR, ignoreCancelled = true)
+    public void onCommandPreprocess(PlayerCommandPreprocessEvent event) {
+
+        final Player player = event.getPlayer();
+        final String cmd = event.getMessage().toLowerCase();
+
+        final String[] cmdArgs = cmd.split(" ");
+        final String primaryCmd = cmdArgs[0].substring(1);
+
+        if (plugin.getConfig().getBoolean("prism.alerts.illegal-commands.enabled")) {
+            if (illegalCommands.contains(primaryCmd)) {
+                final String msg = player.getName() + " attempted an illegal command: " + primaryCmd + ". Originally: "
+                        + cmd;
+                player.sendMessage(Prism.messenger.playerError("Sorry, this command is not available in-game."));
+                plugin.alertPlayers(null, msg);
+                event.setCancelled(true);
+                // Log to console
+                if (plugin.getConfig().getBoolean("prism.alerts.illegal-commands.log-to-console")) {
+                    Prism.log(msg);
+                }
+
+                // Log to commands
+                List<String> commands = plugin.getConfig().getStringList("prism.alerts.illegal-commands.log-commands");
+                MiscUtils.dispatchAlert(msg, commands);
+            }
+        }
+
+        if (!Prism.getIgnore().event("player-command", player)) {
+            return;
+        }
+
+        // Ignore some commands based on config
+        if (ignoreCommands.contains(primaryCmd)) {
+            return;
+        }
+
+        RecordingQueue.addToQueue(ActionFactory.createPlayer("player-command", player, event.getMessage()));
+
+    }
+
+    /**
+     * PlayerJoinEvent.
+     * @param event PlayerJoinEvent
+     */
+    @EventHandler(priority = EventPriority.MONITOR, ignoreCancelled = true)
+    public void onPlayerJoin(final PlayerJoinEvent event) {
+
+        final Player player = event.getPlayer();
+
+        // Lookup player for cache reasons
+        PlayerIdentification.cachePrismPlayer(player);
+
+        // Track the join event
+        if (!Prism.getIgnore().event("player-join", player)) {
+            return;
+        }
+
+        String ip = null;
+        if (plugin.getConfig().getBoolean("prism.track-player-ip-on-join")) {
+            if (player.getAddress() != null) {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [These nested if statements could be combined](https://app.codacy.com/gh/AddstarMC/Prism-Bukkit/pullRequest?prid=5438808)